mp2i/cogs/suggestions.py

```diff
@@ -1,218 +1,391 @@
             update(SuggestionModel)
-            .where(SuggestionModel.message_id == suggestion.id)
+            .where(SuggestionModel.id == suggestion.id)
             .values(
-                state="accepted" if accepted else "declined" if declined else "closed",
-                date=datetime.now()
+                state=new_state.value,
+                handled_by=staff,
+                handled_time=datetime.now()
             )
         )
-        if accepted:
-            embed = discord.Embed(colour=0x77B255, title=f"Suggestion acceptée", description=f"> {citation}\n_**Note**: Il faut parfois attendre plusieurs jours avant qu'elle soit effective_")
-        elif declined:
-            embed = discord.Embed(colour=0xDD2E44, title=f"Suggestion refusée", description=f"> {citation}")
-        else:
-            embed = discord.Embed(colour=0xA9A6A7, title=f"Suggestion fermée", description=f"> {citation}")
-        file = discord.File(STATIC_DIR / "img/alert.png")
-        embed.set_thumbnail(url="attachment://alert.png")
-        embed.set_author(name=suggestion.author.name)
+        await response.send("Suggestion traitée.", ephemeral=True)
+        await thread.edit(locked=True, archived=True)
 
-        await channel.send(file=file, embed=embed)
-        await suggestion.delete()
 
-    @Cog.listener("on_raw_reaction_add")
-    @guild_only()
-    async def add_pin(self, payload) -> None:
+    @Cog.listener("on_message_delete")
+    @has_any_role("Administrateur", "Modérateur")
+    async def on_message_delete(self, message) -> None:
         """
-        Add a pin to a message and send it to website channel when
-        it reach the required number of pins reactions.
+        Delete suggestion in database when message is deleted.
         """
-        if str(payload.emoji) != "📌":
-            return
+        database.execute(
+            delete(SuggestionModel)
+            .where(SuggestionModel.message_id == message.id)
+        )
 
-        channel = self.bot.get_channel(payload.channel_id)
-        message = await channel.fetch_message(payload.message_id)
-        pins = discord.utils.get(message.reactions, emoji="📌", me=False)
-        if pins is None or pins.count < self.MINIMUM_PINS:
+    @hybrid_command(name="create_proposal")
+    @has_any_role("Administrateur")
+    async def create(self, ctx, channel: discord.TextChannel) -> None:
+        """
+        Send the proposal suggestion message
+        """
+        await self.__send_suggestions_process(channel)
+        await ctx.send("Le salon des suggestions a été créé.", ephemeral=True)
+
+    @hybrid_command(name="close")
+    @has_any_role("Administrateur")
+    @choices(
+        state=[
+            Choice(name="Accepter", value=State.ACCEPTED.value),
+            Choice(name="Refuser", value=State.DECLINED.value),
+            Choice(name="Fermer simplement", value=State.CLOSED.value),
+        ]
+    )
+    async def close(self, ctx: Context, state: str) -> None:
```

Comment:
```suggestion
    async def close(self, ctx, state: str) -> None:
```
On n'a jamais mit que c'était un contexte dans les autres fonction, y'a pas trop de raison pour que on l'indique ici